### PR TITLE
Registry Mappings

### PR DESCRIPTION
**Pull request by NoComment1105**
This maps unmapped things in the registry directory

### CODE REVIEW

File: mappings/net/minecraft/util/Holder.mapping [hunk 1]
```diff
@@ -8,6 +8,8 @@ CLASS net/minecraft/unmapped/C_cjzoxshv net/minecraft/util/Holder
 		ARG 1 registryKey
 	METHOD m_hkxploul createDirect (Ljava/lang/Object;)Lnet/minecraft/unmapped/C_cjzoxshv;
 		ARG 0 value
+	METHOD m_igwfqztq unwrap ()Lcom/mojang/datafixers/util/Either;
+	METHOD m_okckgmwt hackyErase (Lnet/minecraft/unmapped/C_cjzoxshv;)Lnet/minecraft/unmapped/C_cjzoxshv;
```
Comment (IotaBread) [inline mappings/net/minecraft/util/Holder.mapping]: This name is very weird

File: mappings/net/minecraft/util/dynamic/Codecs.mapping [hunk 2]
```diff
@@ -15,6 +15,7 @@ CLASS net/minecraft/unmapped/C_tyjpezxh net/minecraft/util/dynamic/Codecs
 		ARG 0 right
 	METHOD m_ezutwlll (Lcom/mojang/serialization/Codec;Ljava/lang/String;Ljava/lang/String;Lcom/mojang/serialization/codecs/RecordCodecBuilder$Instance;)Lcom/mojang/datafixers/kinds/App;
 		ARG 3 instance
+	METHOD m_ihvqsgya retreiveContext (Ljava/util/function/Function;)Lcom/mojang/serialization/MapCodec;
```
Comment (IotaBread) [inline mappings/net/minecraft/util/dynamic/Codecs.mapping]:
```suggestion
	METHOD m_ihvqsgya retrieveContext (Ljava/util/function/Function;)Lcom/mojang/serialization/MapCodec;
```

File: mappings/net/minecraft/util/registry/BuiltinRegistries.mapping [hunk 3]
```diff
@@ -20,7 +20,20 @@ CLASS net/minecraft/unmapped/C_tcccilzh net/minecraft/util/registry/BuiltinRegis
...
+		ARG 1 name
+		ARG 2 o
+	METHOD m_dbdwurdy register (Lnet/minecraft/unmapped/C_tqxyjqsk;Ljava/lang/String;Ljava/lang/Object;)Lnet/minecraft/unmapped/C_cjzoxshv;
+		ARG 0 type
+		ARG 1 name
+		ARG 2 o
+	METHOD m_gefiajzm register (Lnet/minecraft/unmapped/C_tqxyjqsk;Lnet/minecraft/unmapped/C_xhhleach;Ljava/lang/Object;)Lnet/minecraft/unmapped/C_cjzoxshv;
+		ARG 0 type
+		ARG 1 key
+		ARG 2 o
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/BuiltinRegistries.mapping]:
```suggestion
		ARG 2 value
	METHOD m_dbdwurdy register (Lnet/minecraft/unmapped/C_tqxyjqsk;Ljava/lang/String;Ljava/lang/Object;)Lnet/minecraft/unmapped/C_cjzoxshv;
		ARG 0 type
		ARG 1 name
		ARG 2 value
	METHOD m_gefiajzm register (Lnet/minecraft/unmapped/C_tqxyjqsk;Lnet/minecraft/unmapped/C_xhhleach;Ljava/lang/Object;)Lnet/minecraft/unmapped/C_cjzoxshv;
		ARG 0 type
		ARG 1 key
		ARG 2 value
```

File: mappings/net/minecraft/util/registry/BuiltinRegistries.mapping [hunk 4]
```diff
@@ -34,3 +47,7 @@ CLASS net/minecraft/unmapped/C_tcccilzh net/minecraft/util/registry/BuiltinRegis
 		ARG 0 registryRef
 		ARG 1 lifecycle
 		ARG 2 defaultValueSupplier
+	METHOD m_xcdqqnrd register (Lnet/minecraft/unmapped/C_tqxyjqsk;Lnet/minecraft/unmapped/C_ncpywfca;Ljava/lang/Object;)Lnet/minecraft/unmapped/C_cjzoxshv;
+		ARG 0 type
+		ARG 1 id
+		ARG 2 o
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/BuiltinRegistries.mapping]:
```suggestion
		ARG 2 value
```

File: mappings/net/minecraft/util/registry/DynamicRegistryManager.mapping [hunk 5]
```diff
@@ -25,26 +29,83 @@ CLASS net/minecraft/unmapped/C_wqxmvzdq net/minecraft/util/registry/DynamicRegis
...
+		ARG 0 key
+	METHOD m_matjkzip captureMap (Lcom/mojang/serialization/codecs/UnboundedMapCodec;)Lcom/mojang/serialization/Codec;
 	METHOD m_nnjsvctt load (Lcom/mojang/serialization/DynamicOps;Lnet/minecraft/unmapped/C_yftyvswu$C_ogvxrsre;Lnet/minecraft/unmapped/C_wqxmvzdq$C_ydhbsuwm;)V
 		COMMENT Loads elements from the {@code ops} into the registry specified by {@code
 		COMMENT info} within the {@code manager}. Note that the resource manager instance
 		COMMENT is kept within the {@code ops}.
+		ARG 1 bound
 		ARG 2 info
-	METHOD m_physpezp (Lnet/minecraft/unmapped/C_wqxmvzdq$C_qndozwiw;Lnet/minecraft/unmapped/C_nyvkclsy$C_dxgwghhk;Lnet/minecraft/unmapped/C_wqxmvzdq$C_ydhbsuwm;)V
+	METHOD m_physpezp addBuildInElements (Lnet/minecraft/unmapped/C_wqxmvzdq$C_qndozwiw;Lnet/minecraft/unmapped/C_nyvkclsy$C_dxgwghhk;Lnet/minecraft/unmapped/C_wqxmvzdq$C_ydhbsuwm;)V
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/DynamicRegistryManager.mapping]: Is this supposed to be `addBuiltInElements`?

File: mappings/net/minecraft/util/registry/DynamicRegistryManager.mapping [hunk 6]
```diff
@@ -25,26 +29,83 @@ CLASS net/minecraft/unmapped/C_wqxmvzdq net/minecraft/util/registry/DynamicRegis
...
+	METHOD m_vutgtdyk registries ()Ljava/util/stream/Stream;
+	METHOD m_xnavlkkh createFreshRegistries ()Ljava/util/Map;
 	METHOD m_zbdqwvdm register (Lcom/google/common/collect/ImmutableMap$Builder;Lnet/minecraft/unmapped/C_xhhleach;Lcom/mojang/serialization/Codec;Lcom/mojang/serialization/Codec;)V
 		ARG 0 infosBuilder
 		ARG 1 registryRef
 		ARG 2 entryCodec
 		ARG 3 networkEntryCodec
+	METHOD m_zygzorae readFromDisk (Lcom/mojang/serialization/Dynamic;)Lnet/minecraft/unmapped/C_wqxmvzdq;
+	CLASS C_ggmzysrd Frozen
+	CLASS C_gmaxoias ImmutableRegistryAccess
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/DynamicRegistryManager.mapping]: Is the super class of this one a `RegistryManager`? If so, the name should be `ImmutableRegistryManager`

File: mappings/net/minecraft/util/registry/RegistryFixedCodec.mapping [hunk 7]
```diff
@@ -5,5 +5,8 @@ CLASS net/minecraft/unmapped/C_qosrsbdy net/minecraft/util/registry/RegistryFixe
 	METHOD decode decode (Lcom/mojang/serialization/DynamicOps;Ljava/lang/Object;)Lcom/mojang/serialization/DataResult;
 		ARG 1 dynamicOps
 	METHOD encode encode (Ljava/lang/Object;Lcom/mojang/serialization/DynamicOps;Ljava/lang/Object;)Lcom/mojang/serialization/DataResult;
+		ARG 1 holder
 		ARG 2 dynamicOps
+		ARG 3 o
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/RegistryFixedCodec.mapping]:
```suggestion
		ARG 3 value
```

File: mappings/net/minecraft/util/registry/RegistryResourceAccess.mapping [hunk 8]
```diff
@@ -11,10 +11,17 @@ CLASS net/minecraft/unmapped/C_nyvkclsy net/minecraft/util/registry/RegistryReso
 	CLASS C_dxgwghhk InMemoryStorage
 		FIELD f_alenzwoq entries Ljava/util/Map;
 		FIELD f_zcgnytxp LOGGER Lorg/slf4j/Logger;
-		METHOD m_jxopatrk (Lnet/minecraft/unmapped/C_wqxmvzdq;Lnet/minecraft/unmapped/C_xhhleach;Lcom/mojang/serialization/Encoder;ILjava/lang/Object;Lcom/mojang/serialization/Lifecycle;)V
+		METHOD m_jxopatrk add (Lnet/minecraft/unmapped/C_wqxmvzdq;Lnet/minecraft/unmapped/C_xhhleach;Lcom/mojang/serialization/Encoder;ILjava/lang/Object;Lcom/mojang/serialization/Lifecycle;)V
+			ARG 1 manager
+			ARG 2 arg
 			ARG 3 encoder
+			ARG 4 id
+			ARG 5 o
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/RegistryResourceAccess.mapping]:
```suggestion
			ARG 5 value
```

File: mappings/net/minecraft/util/registry/RegistryResourceAccess.mapping [hunk 9]
```diff
@@ -11,10 +11,17 @@ CLASS net/minecraft/unmapped/C_nyvkclsy net/minecraft/util/registry/RegistryReso
 	CLASS C_dxgwghhk InMemoryStorage
 		FIELD f_alenzwoq entries Ljava/util/Map;
 		FIELD f_zcgnytxp LOGGER Lorg/slf4j/Logger;
-		METHOD m_jxopatrk (Lnet/minecraft/unmapped/C_wqxmvzdq;Lnet/minecraft/unmapped/C_xhhleach;Lcom/mojang/serialization/Encoder;ILjava/lang/Object;Lcom/mojang/serialization/Lifecycle;)V
+		METHOD m_jxopatrk add (Lnet/minecraft/unmapped/C_wqxmvzdq;Lnet/minecraft/unmapped/C_xhhleach;Lcom/mojang/serialization/Encoder;ILjava/lang/Object;Lcom/mojang/serialization/Lifecycle;)V
+			ARG 1 manager
+			ARG 2 arg
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/RegistryResourceAccess.mapping]:
```suggestion
			ARG 1 manager
```

File: mappings/net/minecraft/util/registry/SimpleRegistry.mapping [hunk 10]
```diff
@@ -1,6 +1,36 @@
...
+		ARG 1 tag
+	METHOD m_onyhfito getHoldersInOrder ()Ljava/util/List;
+	METHOD m_qtxkrhoa registerMapping (ILnet/minecraft/unmapped/C_xhhleach;Ljava/lang/Object;Lcom/mojang/serialization/Lifecycle;Z)Lnet/minecraft/unmapped/C_cjzoxshv;
+		ARG 1 rawId
+		ARG 2 key
+		ARG 3 entry
+		ARG 4 lifecycle
+	METHOD m_xynptmgp getValueFromNullable (Lnet/minecraft/unmapped/C_cjzoxshv$C_rjzpeyec;)Ljava/lang/Object;
+		ARG 0 reference
+	METHOD m_yxxnntbr validataeWrite (Lnet/minecraft/unmapped/C_xhhleach;)V
```
Comment (IotaBread) [inline mappings/net/minecraft/util/registry/SimpleRegistry.mapping]:
```suggestion
	METHOD m_yxxnntbr validateWrite (Lnet/minecraft/unmapped/C_xhhleach;)V
```

**Review by IotaBread — COMMENTED** [review]
The parameter name `o` is only used for `equals(Object)`

File: mappings/net/minecraft/util/registry/DynamicRegistryManager.mapping (hunk 5 above)
Comment (NoComment1105) [reply mappings/net/minecraft/util/registry/DynamicRegistryManager.mapping]: Yes it is, mb

File: mappings/net/minecraft/util/Holder.mapping (hunk 1 above)
Comment (NoComment1105) [reply mappings/net/minecraft/util/Holder.mapping]: Fixed

File: mappings/net/minecraft/util/registry/RegistryResourceAccess.mapping (hunk 9 above)
Comment (NoComment1105) [reply mappings/net/minecraft/util/registry/RegistryResourceAccess.mapping]: Was a mistake to put `arg` anyway, I replaced this with the correct arg name

**Comment by NoComment1105** [comment]
This PR now maps the new registry things, introduced in 1.18.2-pre1

**Review by LambdAurora — APPROVED** [review]

**Comment by NoComment1105** [comment]
There was only 1 new method for 1.18.2-pre2 here. It has been added now

**Review by IotaBread — APPROVED** [review]

File: mappings/net/minecraft/util/registry/DynamicRegistryManager.mapping [hunk 11]
```diff
@@ -25,26 +29,85 @@ CLASS net/minecraft/unmapped/C_wqxmvzdq net/minecraft/util/registry/DynamicRegis
 		COMMENT
 		COMMENT @throws IllegalStateException if the registry does not exist
 		ARG 1 key
+	METHOD m_dlcbkyln (Lnet/minecraft/unmapped/C_xhhleach;)Lcom/mojang/serialization/DataResult;
+		ARG 0 key
+	METHOD m_esyudkce allElementsLifecycle ()Lcom/mojang/serialization/Lifecycle;
+	METHOD m_fpwtdfne ownedNetworkableRegistries ()Ljava/util/stream/Stream;
+	METHOD m_jffdyfoo retreiveRegistry (Lnet/minecraft/unmapped/C_xhhleach;Lcom/mojang/serialization/Dynamic;)Lnet/minecraft/unmapped/C_tqxyjqsk;
```
Comment (triphora) [inline mappings/net/minecraft/util/registry/DynamicRegistryManager.mapping]: typo
```suggestion
	METHOD m_jffdyfoo retrieveRegistry (Lnet/minecraft/unmapped/C_xhhleach;Lcom/mojang/serialization/Dynamic;)Lnet/minecraft/unmapped/C_tqxyjqsk;
```

File: mappings/net/minecraft/util/registry/Registry.mapping [hunk 12]
```diff
@@ -149,43 +150,96 @@ CLASS net/minecraft/unmapped/C_tqxyjqsk net/minecraft/util/registry/Registry
...
 	METHOD m_mazpxlja get (Lnet/minecraft/unmapped/C_ncpywfca;)Ljava/lang/Object;
 		ARG 1 id
 	METHOD m_mfexjqmi getEntryLifecycle (Ljava/lang/Object;)Lcom/mojang/serialization/Lifecycle;
 		COMMENT Gets the lifecycle of a registry entry.
 		ARG 1 entry
+	METHOD m_mqvnokhq registerDefaulted (Lnet/minecraft/unmapped/C_xhhleach;Ljava/lang/String;Lcom/mojang/serialization/Lifecycle;Lnet/minecraft/unmapped/C_tqxyjqsk$C_umlcfeib;)Lnet/minecraft/unmapped/C_zogerkic;
+		ARG 0 key
+		ARG 1 id
+		ARG 2 lifecycle
+		ARG 3 boostrap
```
Comment (triphora) [inline mappings/net/minecraft/util/registry/Registry.mapping]:
```suggestion
		ARG 3 bootstrap
```

File: mappings/net/minecraft/util/registry/SimpleRegistry.mapping [hunk 13]
```diff
@@ -1,6 +1,34 @@
 CLASS net/minecraft/unmapped/C_secmvxxe net/minecraft/util/registry/SimpleRegistry
+	FIELD f_autwpodl instrusiveHolderCache Ljava/util/Map;
```
Comment (triphora) [inline mappings/net/minecraft/util/registry/SimpleRegistry.mapping]:
```suggestion
	FIELD f_autwpodl intrusiveHolderCache Ljava/util/Map;
```

**Review by triphora — APPROVED** [review]
just typos

**Review by triphora — APPROVED** [review]
There are a couple more usages of `boostrap` that I didn't catch the first time around, just find-and-replace for those and this should be good

**Pull request merged**